schematic/utils/cli_utils.py

```diff
@@ -39,17 +39,17 @@ def extract(dictionary: Any, key: Any) -> Union[Any, None]:
 
 def fill_in_from_config(
     arg_name: str,
-    arg_value: Any,
-    config_keys: Sequence[Any]
+    config_keys: Sequence[Any],
+    arg_value: Any = None
```

Comment:
Set `arg_value` to default to `None` since there's a check to see if it's `None` or not in the beginning of the function body. If we allow it to default to `None` then, we can easily call `fill_in_from_config()` for attributes that are not passed as optional CLI arguments.

Reply:
Here's what I propose:
- Revert the change to the `fill_in_from_config()` function signature (order and `None`default). 
- Create `exceptions.py` submodule with new `MissingConfigValueError` and `MissingConfigAndArgumentValueError` exception classes
- Move `assert config_value is not None` error to within the `query_dict()` function, switch to using `MissingConfigValueError` instead of `AssertionError`, and update the error message accordingly (_i.e._ no reference to argument names)
- Catch `MissingConfigValueError` error when calling `query_dict()` in `fill_in_from_config()` and switch error to `MissingConfigAndArgumentValueError` with the more meaningful error message (_i.e._ with reference to argument name)